src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java

```diff
@@ -89,40 +91,80 @@ public void update()
             {
                 if (i == index)
                 {
-                    if (buildingEntry.getValue() instanceof AbstractBuildingWorker
-                          && !list.contains(buildingEntry.getValue())
-                          && ((AbstractBuildingWorker) buildingEntry.getValue()).needsAnything())
+                    if(buildingEntry.getValue() instanceof AbstractBuildingWorker
+                            && !list.contains(buildingEntry.getValue())
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "contains()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)